src/uct/sm/self/self.c

```diff
@@ -8,21 +8,23 @@
 #include <uct/sm/base/sm_ep.h>
 #include <ucs/type/class.h>
 #include <ucs/sys/string.h>
+#include <ucs/arch/cpu.h>
 #include "self.h"
 
 
 #define UCT_SELF_NAME "self"
 
 #define UCT_SELF_IFACE_SEND_BUFFER_GET(_iface) \
-    { \
-        if ((_iface)->send_buffer == NULL) { \
-            (_iface)->send_buffer = ucs_malloc((_iface)->send_size, \
-                                               "self_send_buffer"); \
-            if ((_iface)->send_buffer == NULL) { \
+    ({ \
+        void *ptr = ucs_mpool_get_inline(&(_iface)->msg_mp); \
+        if (ucs_unlikely(ptr == NULL)) { \
                 return UCS_ERR_NO_MEMORY; \
-            } \
         } \
-    }
+        ptr; \
+    })
+
+#define UCT_SELF_IFACE_SEND_BUFFER_PUT(_buffer) \
```

Comment:
not sure we need helper macro for single-line function..

Reply:
removed